server/src/services/embedding.ts

```diff
@@ -1,104 +1,179 @@
+// Gemini configuration
+const GEMINI_API_KEY = process.env.GEMINI_API_KEY || '';
+if (!GEMINI_API_KEY) {
+  console.warn('GEMINI_API_KEY is not set in environment variables. Please add it to your .env file.');
+}
+
+const GEMINI_VECTOR_SIZE = parseInt(process.env.GEMINI_VECTOR_SIZE || '768', 10);
 
 // Create fallback service for embedding operations
 const embeddingFallback = new FallbackService('Embedding');
 
 /**
  * Generate an embedding vector for the provided text
  * 
- * Uses Qdrant's built-in FastEmbed capability
+ * Uses Google's Gemini for embeddings
  * 
  * @param text The text to generate an embedding for
+ * @param taskType Optional task type for Gemini embeddings ('retrieval_document' or 'retrieval_query')
  * @returns A vector representation of the text
  */
-export const createEmbedding = async (text: string): Promise<number[]> => {
+export const createEmbedding = async (text: string, taskType?: 'retrieval_document' | 'retrieval_query'): Promise<number[]> => {
   return embeddingFallback.withFallback(
     'generateVector',
     // Fallback function - generate random vector
     () => {
       // Log information about text being embedded to help debugging
       console.debug(`Generated fallback embedding for text of length ${text.length}.`);
-      return Array.from({ length: VECTOR_SIZE }, () => Math.random() - 0.5);
+      return Array.from({ length: GEMINI_VECTOR_SIZE }, () => Math.random() - 0.5);
     },
-    // Primary function - call embedding API
+    // Primary function - call Gemini embedding API
     async () => {
-      // Using Qdrant's server-side FastEmbed integration
-      const headers: Record<string, string> = { 'Content-Type': 'application/json' };
-      
-      if (QDRANT_API_KEY) {
-        headers['api-key'] = QDRANT_API_KEY;
-      }
-      
-      const response = await axios.post(
-        `${QDRANT_URL}/embeddings`,
-        {
-          text: text,
-          model: 'fastembed', // Use the default FastEmbed model
-        },
-        { headers }
-      );
-      
-      return response.data.embedding;
+      return generateGeminiEmbedding(text, taskType);
     },
     embeddingFallback.isFallbackActive()
   );
 };
 
 /**
- * Helper function to calculate cosine similarity between two vectors
+ * Generate embedding using Google's Gemini API
  * 
- * @param a First vector
- * @param b Second vector
- * @returns Cosine similarity score (0-1)
+ * @param text Text to generate embedding for
+ * @param taskType Task type: 'retrieval_document' for indexed docs, 'retrieval_query' for search
+ * @returns Vector representation of the text
  */
-export function cosineSimilarity(a: number[], b: number[]): number {
-  if (a.length !== b.length) {
-    throw new Error('Vectors must be of the same length');
+async function generateGeminiEmbedding(
+  text: string, 
+  taskType: 'retrieval_document' | 'retrieval_query' = 'retrieval_document'
+): Promise<number[]> {
```

Comment:
here as well, we can use enum for type safe

```suggestion
async function generateGeminiEmbedding(
  text: string, 
  taskType: TaskType = TaskType.RETRIEVAL_DOCUMENT
): Promise<number[]> {
```